### DISCUSSION TITLE
Flet-pyodide base-url must work automatically on any path.

### DISCUSSION THREAD

**Discussion by SKbarbon** [flet-dev/flet#1392] [Ideas]
Currently, `flet-pyodide` dist folder is generated by a simple command called `flet publish`.

But the problem is, if you want to put this `dist` folder in a specific path on your webhost, you need to specify it by `--base-url`, then this set a `<base href="/">` based on `--base-url` value. This can be easier if the javascript set the base-url to be set automatically. I did edit the `<base href` on `index.html` of the dist file to make it choosen automatic, and it did work, this is what I did:

```html
<!DOCTYPE html>
<html>

<head>
  <!-- Here is the replacement of <base href=""> -->
  <script>
    const head = document.getElementsByTagName('head')[0];
    const base = document.createElement('base');
    base.href = window.location.origin + window.location.pathname;
    head.appendChild(base);
  </script>

  <meta charset="UTF-8">
  <meta content="IE=Edge" http-equiv="X-UA-Compatible">
  <meta name="description" content="Flet application.">
```

So, what the advantages of using this:
- more simplicity.
- Can put it on any path without requiring to rePublish it to update the `--base-url`.

Can `flet` use this ?

**Comment by FeodorFitsner** [comment]
Hm, that works in one specific case only: statically published web site with `#` used for routing. Otherwise, visiting a page with a routing would use a wrong path. For example, going to https://flet-controls-gallery.fly.dev/buttons/elevatedbutton will put `/buttons/elevatedbutton` to base href which is incorrect base path for resources.
But the idea is definitely nice!
